packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -274,6 +275,13 @@ export interface FunctionProps extends FunctionOptions {
    * the handler.
    */
   readonly handler: string;
+
+  /**
+   * The filesystem configuration for the lambda function
+   *
+   * @default - will not mount any filesystem
+   */
+  readonly filesystems?: FileSystemOptions;
```

Comment:
Would it be simpler to just ask for a list of `IAccessPoint` and `mountPath`?

```ts
readonly filesystems?: FileSystem[];

interface FileSystem {
  readonly accessPoint: efs.IAccessPoint;
  readonly mountPath: string;
}
```

Reply:
Yes it would. But I think we might support other filesystem types in addition to efs. What do you think?

Reply:
I see. That's a good point.

Then how about an API like this -

```ts
const accessPoint: efs.IAccessPoint = ...

new lambda.Function(this, 'Function',  {
  ...
  filesystems: [{
    target: new AccessPointFilesystemTarget(accessPoint),
    mountPath: '...',
  }]
```

This is similar to how we have done event sources in lambda, so you can follow the same pattern (see [`IEventSource`](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-lambda/lib/event-source.ts) and its [implementations](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-lambda-event-sources/lib)).
The starting point will be something like -

```ts
interface Filesystem {
  readonly target: IFilesystemTarget
  readonly mountPath: string;
}

class AccessPointFilesystemTarget implements IFilesystemTarget {
  ...
}
```

Unlike lambda event sources, I don't think we need a separate package in the case of file system. All of it can stay within `aws-lambda`. @eladb - WDYT?

Reply:
Hi @nija-at 

Actually, all we need is simply the access point arn and it seems we dont need to `bind` the function. 

```ts
interface IFilesystemTarget {
  bind(handler: Function): FilesystemTargetConfig;
}
```

To make it even simpler, I proposed the API like this:


```ts
 export interface FileSystem {
  /**
   * the mount target of the filesystem
   */
  readonly target: IFilesystemTarget;

  /**
   * the target directory in the lambda runtime environment
   */
  readonly mountPath: string;
}

/**
 * Represents a filesystem target containing the resource ARN
 */
export interface IFilesystemTarget {
  /**
   * ARN of the target
   */
  readonly targetArn: string;
}

/**
 * Represents the mount target of the Amazon EFS access point
 * @param accessPoint the Amazon EFS access point
 */
export class EfsAccessPointTarget implements IFilesystemTarget {
  constructor(readonly accessPoint: efs.IAccessPoint) {
  }
  public get targetArn(): string {
    return this.accessPoint.accessPointArn
  };
}
```
So we can configure the filesystems for the lambda.Function like this

```ts
  filesystems: [{
    target: new lambda.EfsAccessPointTarget(accessPoint),
    mountPath:  '/mnt/msg',
  }],
```

When we support new filesystems we can configure it  like this

```ts
  filesystems: [
     {
         target: new lambda.EfsAccessPointTarget(efsAccessPoint),
         mountPath:  '/mnt/msg',
     },
     {
         target: new lambda.LustreAccessPointTarget(lustreAccessPoint),
         mountPath:  '/mnt/msg2',
     },
     {
         target: new lambda.WhateverAccessPointTarget(whateverAccessPoint),
         mountPath:  '/mnt/msg3,
     },
  ],
```

What do you think?